src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/Splinter.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.*;
+
+public class Splinter extends CustomEnchantment implements EnchantmentAbility {
+
+    private final HoloItemsRevamp plugin;
+
+    private final MaterialSetTag COMPATIBLE_MATERIALS;
+
+    private static final Map<Player, ActiveSplinter> splinters = new HashMap<>();
+
+    public Splinter(HoloItemsRevamp plugin){
+        super(plugin, "splinter");
+        this.plugin = plugin;
+        var materialSetTagKey = new NamespacedKey(plugin, "splinter_materials");
+        this.COMPATIBLE_MATERIALS = new MaterialSetTag(materialSetTagKey, Splinter::isCompatibleMaterial).lock();
+    }
+
+    @Override
+    public int getMaxLevel() {
+        return 1;
+    }
+
+    @Override
+    public boolean conflictsWith(@NotNull Enchantment enchantment) {
+        return false;
+    }
+
+    @Override
+    public boolean canEnchantItem(@NotNull ItemStack itemStack) {
+        return MaterialTags.AXES.isTagged(itemStack);
+    }
+
+    @Override
+    public @NotNull Component displayName(int i) {
+        return Component.text()
+            .color(NamedTextColor.GRAY)
+            .decoration(TextDecoration.ITALIC, false)
+            .append(Component.text("Splinter"))
+            .build();
+    }
+
+    @Override
+    public int getCostMultiplier() {
+        return 12;
+    }
+
+    @Override
+    public void onBlockBreak(BlockBreakEvent event, ItemStack itemStack) {
+        Block blockBroken = event.getBlock();
+        if(isInvalidSplinterType(blockBroken.getType()) || isInvalidSplinterLocation(blockBroken.getLocation())){
+            return;
+        }
+        // Presuming the block is a log - ie, it has an orientation.
+        tryCleanupSplinterData(event.getPlayer(), blockBroken);
+
+        // Check and handle if the block is a trunk
+        // TODO
+
+        // Check and handle if the block is a branch
+        // TODO
+    }
+
+    /**
+     * Try to cleanup a player's Splinter data. If the data is scheduled for cleanup, but the newTrunk isn't a valid
+     * new trunk block, it will delete the data from the map entirely.
+     * @param player The player's
+     * @param newTrunk The proposed new trunk block
+     */
+    private void tryCleanupSplinterData(Player player, Block newTrunk){
+        final var activeSplinter = splinters.get(player);
+        // if(!shouldCleanup)
+        if(activeSplinter != null && activeSplinter.scheduledSplinters != 0){
+            // there is still a splinter going, so don't cleanup until that one's done
+            return;
+        }
+
+        splinters.remove(player);
+
+        BlockData data = newTrunk.getBlockData();
+        if(!(data instanceof Orientable)){
+            // all logs are orientable, so this isn't a log
+            // TODO: ... mushrooms?
+            return;
+        }
+
+        Axis orientationAxis = ((Orientable) data).getAxis();
+        if(orientationAxis != Axis.Y){
+            return;
+        }
```

Comment:
Being so verbose hurts readability
```suggestion
        if(!(newTrunk.getBlockData() instanceof Orientable orientable)){
            // all logs are orientable, so this isn't a log
            // TODO: ... mushrooms?
            return;
        }

        final var orientationAxis = orientable.getAxis();
        if(orientationAxis != Axis.Y){
            return;
        }
```

Reply:
It's pattern matching for instanceof which was a feature delivered in jdk 16.
https://openjdk.org/jeps/394